<div class='alert alert-warning'>

SciPy's interactive examples with Jupyterlite are experimental and may not always work as expected. Execution of cells containing imports may result in large downloads (up to 60MB of content for the first import from SciPy). Load times when importing from SciPy may take roughly 10-20 seconds. If you notice any problems, feel free to open an [issue](https://github.com/scipy/scipy/issues/new/choose).

</div>

We follow the example of table 3 in [1]: Fourteen children were divided
randomly into two groups. Their ranks at performing a specific tests are
as follows.


In [ ]:
import numpy as np
x = [1, 2, 3, 4, 6, 7, 8]
y = [5, 9, 10, 11, 12, 13, 14]

We use the BWS test to assess whether there is a statistically significant
difference between the two groups.
The null hypothesis is that there is no difference in the distributions of
performance between the two groups. We decide that a significance level of
1% is required to reject the null hypothesis in favor of the alternative
that the distributions are different.
Since the number of samples is very small, we can compare the observed test
statistic against the *exact* distribution of the test statistic under the
null hypothesis.


In [ ]:
from scipy.stats import bws_test
res = bws_test(x, y)
print(res.statistic)

5.132167152575315

This agrees with $B = 5.132$ reported in [1]. The *p*-value produced
by `bws_test` also agrees with $p = 0.0029$ reported in [1].


In [ ]:
print(res.pvalue)

0.002913752913752914

Because the p-value is below our threshold of 1%, we take this as evidence
against the null hypothesis in favor of the alternative that there is a
difference in performance between the two groups.